components/scream/scripts/test_all_scream.py

```diff
@@ -66,7 +70,7 @@ def __init__(self, cxx, kokkos=None, submit=False, parallel=False, fast_fail=Fal
                 elif self._integration_test:
                     self._baseline_ref = "origin/master"
                     if get_current_commit() != get_current_commit(commit="origin/master"):
```

Comment:
We could move this check (that the HEAD and merge commits are different) into merge_git_ref.

Reply:
Do we even need to check? Merging a ref into itself is a no-op...

Reply:
Maybe not. It depends on what the return code is if it does nothing.

Reply:
I just tried. The return code is 0.